nlp/2021_06_29_long_range_transformers/LongRangeTransformers.ipynb

```diff
@@ -0,0 +1,1940 @@
+    "\n",
+    "🛎️ Disclaimer: this will take some time, you could also opt for downloading our final results by running the following cell and skipping the training phases. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 📥 Importing results"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!git clone https://github.com/ml6team/quick-tips.git\n",
+    "!cd quick-tips && git checkout feature/long_seq_transformers\n",
+    "!mv quick-tips/nlp/2021_06_29_long_range_transformers/models ./models\n",
+    "!rm -rf quick-tips"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 💪 Training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "id": "PJ6mWoPjVstv"
+   },
+   "outputs": [],
+   "source": [
+    "def compute_metrics(pred):\n",
+    "    labels = pred.label_ids\n",
+    "    preds = pred.predictions.argmax(-1)\n",
+    "    precision, recall, f1, _ = precision_recall_fscore_support(labels, preds, average='binary')\n",
+    "    acc = accuracy_score(labels, preds)\n",
+    "    return {\n",
+    "        'accuracy': acc,\n",
+    "        'f1': f1,\n",
+    "        'precision': precision,\n",
+    "        'recall': recall\n",
+    "    }\n",
+    "\n",
+    "def train_and_evaluate(model, identifier, tokenizer, train_ds, test_ds, max_length=1024):\n",
+    "    def tokenization(batched_text):\n",
+    "        return tokenizer(batched_text['text'], padding='max_length', truncation=True, max_length=max_length)\n",
+    "\n",
+    "    # tokenizing both the training and test dataset \n",
+    "    train_ds = train_ds.map(\n",
+    "        tokenization, batched=True, batch_size=len(train_ds), remove_columns=['text'])\n",
+    "    test_ds = test_ds.map(tokenization, batched=True,\n",
+    "                          batch_size=len(test_ds),  remove_columns=['text'])\n",
+    "    train_ds.set_format(\n",
+    "        'torch', columns=['input_ids', 'attention_mask', 'label'])\n",
+    "    test_ds.set_format('torch', columns=[\n",
+    "                       'input_ids', 'attention_mask', 'label'])\n",
+    "\n",
+    "\n",
+    "    parameters = model.num_parameters()\n",
+    "    training_args = TrainingArguments(\n",
+    "        num_train_epochs=5,\n",
+    "        per_device_train_batch_size=4,\n",
+    "        gradient_accumulation_steps=2,\n",
+    "        per_device_eval_batch_size=4,\n",
+    "        evaluation_strategy=\"steps\",\n",
+    "        eval_steps=100, \n",
+    "        disable_tqdm=False,\n",
+    "        load_best_model_at_end=True,\n",
+    "        metric_for_best_model='accuracy', # default value is validation loss, we want the model with the highest accuracy \n",
+    "        warmup_steps=200,\n",
+    "        logging_steps=100,\n",
+    "        learning_rate=5e-6,\n",
+    "        fp16=True,\n",
```

Comment:
did you notice a speedup with or without this?

Reply:
Definitely, however, this doesn't work on Google Colab so maybe we should exclude it?